book/5-data-mgmt-exp-proc/eda-cv-tasks.md

```diff
@@ -0,0 +1,117 @@
+
+### Image Similarity
+1. PCA or ISOMAP per channel can be performed on the images to see how the images are related to each other.
+
+![title](../assets/images/screenshots/isomap_pca.png)
+
+## Annotations (Basic Overview)
+
+### Bounding Boxes Formats
+
+1. COCO - [x_min, y_min, width, height]
+1. YOLO - [x_center, y_center, width, height]
+1. Pascal VOC - [x_min, y_min, x_max, y_max]
+
+For example, if the given format is in Pascal VOC, and the model requires it in YOLO format, we would need to convert the annotations to YOLO format for training.  
+
+An example for during inference: Read images using CV2, run YOLOv3 model, convert and return in the desired format according to what is required to be delivered.
+
+### Masks Format
+
+Masks annotations gives the pixel-wise positions of the mask.
+
+Masks are provided in an array of x-y pairs. It will typically be:
+
+```
+masks_per_instance = [x_1, y_1, x_2, y_2, ... x_n, y_n]
+```
+
+Masks may also come in the form of PNG files, where the PNG's image size would correspond to the original image size. 
+
+## Annotations EDA
+
+### Basic Annotations Analysis
+
+The basic annotations analysis attempts to find characteristics of the annotation such as number (and proportion) of classes as well as the values of the bounding boxes/ masks. 
+
+This is useful because it helps determine what type of metrics should be used, how the data (images) should be split. For metrics, the usual case for this is error analysis of model performance on bounding box sizes. For example, if the bboxes/masks are mostly large objects among all the images, assuming images are sufficiently clear, it is clear that the model will do well. However, if the annotations are small, then the model will inevitably struggle. Hence, the AI Engineer will pay attention to certain metrics more, such as mAP with Area=Large.
+
+The data split refers to how one might partition the data for training/validating/testing the model. Generally, all sets will need equal proportions of the bbox size (small/medium/large) and class distribution. In the ideal scenario, these proportions are easy to split at random, but it is extremely rare. Thus, by performing EDA on the annotations, the Engineer will have a better idea on how the data split can be done. 
+
+With these considerations in mind, these are some guiding questions when performing EDA on annotations: 
+1. How many instances of each classes are there?
+1. Bounding Box - what is width and height distribution overall and per class?
+1. Bounding Box - what is the overall size (W*H) overall and per class?
+1. Bounding Box - What are the top few Width/Height combination? This can be found using clustering methods.
+1. Segmentation - what is distribution of the mask pixel counts overall and per class?
+
```

Comment:
Not sure if space permits - if so, may be nice to put in an example of a height vs width scatterplot. We can't use data from our projects, so would have to find a public one. I found one in this paper: https://www.hindawi.com/journals/sp/2022/5755265/
(will have to attribute to the site if we use it)